main.js

```diff
@@ -126,44 +126,42 @@ class Covid19 extends utils.Adapter {
 
 				// Write Top 5
 				this.log.debug('Top 5 Countries : ' + JSON.stringify(values.slice(0, 5)));
-				const top_Arrary = values.slice(0, 5);
-				let count = 1;
-				for (const i in top_Arrary) {
-					if (count <= 5 ) {
-						let country = top_Arrary[i].country;
-
-						await this.extendObjectAsync('country_Top_5.' + count, {
-							type: 'channel',
-							common: {
-								name: 'Rank ' + count + ' : ' + country,
-							},
-							native: {},
-						});
-
-						country = country.replace(/\s/g, '_');
-						country = country.replace(/\./g, '');
-						this.log.debug('Country loop rank : ' + count + ' ' + JSON.stringify(country));
-						for (const y in values[i]) {
-							if (y !== 'countryInfo') {
-								await this.localCreateState('country_Top_5.' + count + '.' + y, y, values[i][y]);
-							} else {
-								// Only take the flag from country info
-								await this.localCreateState('country_Top_5.' + count + '.flag', 'flag', values[i][y].flag);
-							}
+				for (let position = 1; position <= 5; position++) {
+					const dataset = values[position - 1]; // start at 0
+					let country = dataset.country;
+
+					const channelName = 'country_Top_5.' + position;
```

Comment:
I'm not sure what you were trying to do here. Just use a normal for-loop from 1 to 5 and read the original array :)